### PR TITLE
feat: Add Support for WebView.InvokeScriptAsync

### LINKED ISSUE #3518: Support for WebView.InvokeScriptAsync
## What would you like to be added:
Support for WebView.InvokeScriptAsync on non-Windows platforms: https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.webview.invokescriptasync?view=winrt-19041

## Why is this needed:
This method is currently missing on macOS.  It is a useful method of the WebView control.  Visual Studio for Mac displays the following message when referencing the InvokeScriptAsyncMethod: _"Windows.UI.Xaml.Controls.WebView.InvokeScriptAsync(String, IEnumerable<String>) is not implemented in Uno"_

## For which Platform:

- [ ] iOS
- [ ] Android
- [ ] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

## Anything else we need to know?

There are other WebView methods and events, such as the DOMContentLoaded event, that are not implemented on all platforms.  It would be usefully to have a full implementation of this object.

### PR DESCRIPTION
**Pull request by ajpinedam**
GitHub Issue (If applicable): #

closes #3518 

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

One of the signatures for WebView.InvokeScriptAsync is not supported.

## What is the new behavior?

WebView.InvokeScriptAsync is now supported for iOS, macOS and Android

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information
There was a method with another signature implemented:

`Task<string> InvokeScriptAsync(CancellationToken ct, string script, string[] arguments);`

And this was definitely an option for the users. Unfortunately, the method that was not implemented was the first one to show on the IntelliSense making the users believe that none of the methods were implemented.

Internal Issue (If applicable):

### CODE REVIEW

**Review by davidjohnoliver — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/WebView/WebView.Android.cs [hunk 1]
```diff
@@ -152,6 +153,10 @@ public async Task<string> InvokeScriptAsync(CancellationToken ct, string script,
 			return await tcs.Task;
 		}
 
+		public IAsyncOperation<string> InvokeScriptAsync(string scriptName, IEnumerable<string> arguments) =>
+			InvokeScriptAsync(CancellationToken.None, scriptName, arguments?.ToArray()).AsAsyncOperation();
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/WebView/WebView.Android.cs:157]: It's best if you propagate the `CancellationToken` here.
```suggestion
			AsyncOperation.FromTask(async ct => InvokeScriptAsync(ct, scriptName, arguments?.ToArray()).AsAsyncOperation();
```

File: src/Uno.UI/UI/Xaml/Controls/WebView/WebView.iOSmacOS.cs [hunk 2]
```diff
@@ -72,6 +73,9 @@ public async Task<string> InvokeScriptAsync(CancellationToken ct, string script,
 			return await _nativeWebView.EvaluateJavascriptAsync(ct, string.Format(CultureInfo.InvariantCulture, "javascript:{0}(\"{1}\")", script, argumentString));
 		}
 
+		public IAsyncOperation<string> InvokeScriptAsync(string scriptName, IEnumerable<string> arguments) =>
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/WebView/WebView.iOSmacOS.cs:76]: Same here.

**Review by jeromelaban — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/WebView/WebView.iOSmacOS.cs (hunk 2 above)
Comment (ajpinedam) [reply src/Uno.UI/UI/Xaml/Controls/WebView/WebView.iOSmacOS.cs:76]: Updated

**Comment by jeromelaban** [comment]
@ajpinedam build failure looks like a conversion issue on iOS.

**Pull request merged**